engines/query-sparql/test/QuerySparql-test.ts

```diff
@@ -484,6 +484,72 @@ describe('System test: QuerySparql', () => {
+          // besides the value in the cache.
+          query = `SELECT (strlen(?x) AS ?len) WHERE {
+            ?s ?p ?x
+          }
+          `;
+          stringType = DF.namedNode('http://www.w3.org/2001/XMLSchema#string');
+          store = new Store();
+          quads = [
+            DF.quad(DF.namedNode(':a'), DF.namedNode(':p'), DF.literal('apple', stringType)) ];
+          store.addQuads(quads);
+        });
+
+        it('is used when provided', async() => {
+          const context = <any> { sources: [ store ]};
+          const functionArgumentsCache = {};
+          context.functionArgumentsCache = functionArgumentsCache;
+
+          const bindingsStream = await engine.queryBindings(query, context);
+          expect((await bindingsStream.toArray()).map(res => res.get(DF.variable('len'))!.value)).toEqual(
+            quads.map(q => String(q.object.value.length)),
+          );
+          expect(Object.keys(functionArgumentsCache)).toContain('strlen');
+        });
+
+        it('is used when defaulted', async() => {
+          const alternativeEngine = new QueryEngine();
+          const context = <any> { sources: [ store ]};
+
+          // This combersom way is needed because evaluating with mock with isCalledWith gives you the refference
+          // to the cache and will make it seem like it was filled in from the beginning.
+          const original_function = (<any> alternativeEngine).actorInitQuery.mediatorContextPreprocess.mediate;
+          const emptyCacheMock = jest.fn((arg: { context: IActionContext }) => {
+            expect(arg.context.get(KeysInitQuery.functionArgumentsCache)).toEqual({});
+            return original_function(arg);
+          });
+          (<any> alternativeEngine).actorInitQuery.mediatorContextPreprocess.mediate = emptyCacheMock;
+
+          // Evaluate query once
+          const firstBindingsStream = await alternativeEngine.queryBindings(query, context);
+          expect((await firstBindingsStream.toArray()).map(res => res.get(DF.variable('len'))!.value)).toEqual(
+            quads.map(q => String(q.object.value.length)),
+          );
+
+          const fullCacheMock = jest.fn((arg: { context: IActionContext }) => {
+            expect(arg.context.get(KeysInitQuery.functionArgumentsCache)).not.toBeUndefined();
+            expect(Object.keys(arg.context.get(KeysInitQuery.functionArgumentsCache)!)).toContain('strlen');
+            return original_function(arg);
+          });
+          (<any> alternativeEngine).actorInitQuery.mediatorContextPreprocess.mediate = fullCacheMock;
+          // Evaluate query a second time
+          const secondBindingsStream = await alternativeEngine.queryBindings(query, context);
+          expect((await secondBindingsStream.toArray()).map(res => res.get(DF.variable('len'))!.value)).toEqual(
+            quads.map(q => String(q.object.value.length)),
+          );
+
```

Comment:
Can we also save the functionArgumentsCache value from the context from the two query execs, and compare them at the end of the test?

Something like `expect(firstFuncArgsCache).toBe(secondFuncArgsCache);`